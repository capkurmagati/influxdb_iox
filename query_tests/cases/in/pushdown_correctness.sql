-- Test the correctness of queries when predicates are pushed down
-- IOX_SETUP: TwoMeasurementsPredicatePushDown

-- Test 1: Select everything
SELECT * from restaurant;

-- Test 2: One push-down expression: count > 200
SELECT * from restaurant where count > 200;

-- Test 3: Two push-down expression: count > 200 and town != 'tewsbury'
SELECT * from restaurant where count > 200 and town != 'tewsbury';

-- Test 4: Still two push-down expression: count > 200 and town != 'tewsbury'
SELECT * from restaurant where count > 200 and town != 'tewsbury' and (system =5 or town = 'lawrence');

-- Test 5: three push-down expression: count > 200 and town != 'tewsbury' and count < 40000
SELECT * from restaurant where count > 200 and town != 'tewsbury' and (system =5 or town = 'lawrence') and count < 40000;

-- Test 6: two push-down expression: count > 200 and count < 40000
SELECT * from restaurant where count > 200  and count < 40000;

-- Test 7: two push-down expression on float: system > 4.0 and system < 7.0
SELECT * from restaurant where system > 4.0 and system < 7.0;

-- Test 8: two push-down expression on float: system > 5.0 and system < 7.0
SELECT * from restaurant where system > 5.0 and system < 7.0;

-- Test 9: three push-down expression: system > 5.0 and town != 'tewsbury' and system < 7.0
SELECT * from restaurant where system > 5.0 and town != 'tewsbury' and 7.0 > system;

-- Test 10: three push-down expression: system > 5.0 and town != 'tewsbury' and system < 7.0
SELECT * from restaurant where system > 5.0 and 'tewsbury' != town and system < 7.0 and (count = 632 or town = 'reading');

-- Test 11: four push-down expression: system > 5.0 and town != 'tewsbury' and system < 7.0 and
SELECT * from restaurant where 5.0 < system and town != 'tewsbury' and system < 7.0 and (count = 632 or town = 'reading') and time > to_timestamp('1970-01-01T00:00:00.000000130+00:00');

--TODO: Hit stackoverflow in DF. Ticket https://github.com/apache/arrow-datafusion/issues/419
-- Test 12: three push-down expression: system > 5.0 and town != 'tewsbury' and system < 7.0 and town = 'reading'
-- SELECT * from restaurant where system > 5.0 and 'tewsbury' != town and system < 7.0 and town = 'reading'",
-- let expected = vec![
-- "+-------+--------+-------------------------------+---------+",
-- "| count | system | time                          | town    |",
-- "+-------+--------+-------------------------------+---------+",
-- "| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading |",
-- "+-------+--------+-------------------------------+---------+",

-- Test 13: three push-down expression: system > 5.0 and system < 7.0 and town = 'reading'
SELECT * from restaurant where system > 5.0 and system < 7.0 and town = 'reading';
