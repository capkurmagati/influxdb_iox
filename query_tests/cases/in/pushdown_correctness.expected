-- Test Setup: TwoMeasurementsPredicatePushDown
-- SQL: SELECT * from restaurant;
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 40000 | 5      | 1970-01-01 00:00:00.000000100 | andover   |
| 189   | 7      | 1970-01-01 00:00:00.000000110 | bedford   |
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading   |
| 471   | 6      | 1970-01-01 00:00:00.000000110 | tewsbury  |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where count > 200;
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 40000 | 5      | 1970-01-01 00:00:00.000000100 | andover   |
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading   |
| 471   | 6      | 1970-01-01 00:00:00.000000110 | tewsbury  |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where count > 200 and town != 'tewsbury';
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 40000 | 5      | 1970-01-01 00:00:00.000000100 | andover   |
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading   |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where count > 200 and town != 'tewsbury' and (system =5 or town = 'lawrence');
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 40000 | 5      | 1970-01-01 00:00:00.000000100 | andover   |
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where count > 200 and town != 'tewsbury' and (system =5 or town = 'lawrence') and count < 40000;
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where count > 200  and count < 40000;
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading   |
| 471   | 6      | 1970-01-01 00:00:00.000000110 | tewsbury  |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where system > 4.0 and system < 7.0;
+-------+--------+-------------------------------+-----------+
| count | system | time                          | town      |
+-------+--------+-------------------------------+-----------+
| 40000 | 5      | 1970-01-01 00:00:00.000000100 | andover   |
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence  |
| 372   | 5      | 1970-01-01 00:00:00.000000100 | lexington |
| 632   | 5      | 1970-01-01 00:00:00.000000120 | reading   |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading   |
| 471   | 6      | 1970-01-01 00:00:00.000000110 | tewsbury  |
+-------+--------+-------------------------------+-----------+
-- SQL: SELECT * from restaurant where system > 5.0 and system < 7.0;
+-------+--------+-------------------------------+----------+
| count | system | time                          | town     |
+-------+--------+-------------------------------+----------+
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading  |
| 471   | 6      | 1970-01-01 00:00:00.000000110 | tewsbury |
+-------+--------+-------------------------------+----------+
-- SQL: SELECT * from restaurant where system > 5.0 and town != 'tewsbury' and 7.0 > system;
+-------+--------+-------------------------------+----------+
| count | system | time                          | town     |
+-------+--------+-------------------------------+----------+
| 872   | 6      | 1970-01-01 00:00:00.000000110 | lawrence |
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading  |
+-------+--------+-------------------------------+----------+
-- SQL: SELECT * from restaurant where system > 5.0 and 'tewsbury' != town and system < 7.0 and (count = 632 or town = 'reading');
+-------+--------+-------------------------------+---------+
| count | system | time                          | town    |
+-------+--------+-------------------------------+---------+
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading |
+-------+--------+-------------------------------+---------+
-- SQL: SELECT * from restaurant where 5.0 < system and town != 'tewsbury' and system < 7.0 and (count = 632 or town = 'reading') and time > to_timestamp('1970-01-01T00:00:00.000000130+00:00');
++
++
-- SQL: SELECT * from restaurant where system > 5.0 and system < 7.0 and town = 'reading';
+-------+--------+-------------------------------+---------+
| count | system | time                          | town    |
+-------+--------+-------------------------------+---------+
| 632   | 6      | 1970-01-01 00:00:00.000000130 | reading |
+-------+--------+-------------------------------+---------+
